packages/client/hooks/useUsageSnackNag.ts

```diff
@@ -0,0 +1,77 @@
+import graphql from 'babel-plugin-relay/macro'
+import ms from 'ms'
+import {useEffect} from 'react'
+import {useLazyLoadQuery} from 'react-relay'
+import {RouterProps, useHistory} from 'react-router'
+import Atmosphere from '../Atmosphere'
+import SendClientSegmentEventMutation from '../mutations/SendClientSegmentEventMutation'
+import {TierEnum} from './../__generated__/NewTeamOrgPicker_organizations.graphql'
+import {useUsageSnackNagQuery} from './../__generated__/useUsageSnackNagQuery.graphql'
+import useAtmosphere from './useAtmosphere'
+
+const NAG_EVERY = ms('3m')
+
+const getIsNaggingPath = (history: RouterProps['history']) => {
+  const {location} = history
+  const {pathname} = location
+  return !(pathname.includes('/usage') || pathname.includes('/meet/'))
+}
+
+const shouldNag = (tier: TierEnum, suggestedTier: TierEnum | null) => {
+  if (!suggestedTier) return false
+  const suggestPro = suggestedTier === 'pro' && tier === 'personal'
+  const suggestEnterprise = suggestedTier === 'enterprise' && tier !== 'enterprise'
+  return suggestPro || suggestEnterprise
+}
+
+const emitNag = (atmosphere: Atmosphere, history: RouterProps['history']) => {
+  atmosphere.eventEmitter.emit('addSnackbar', {
+    key: 'usage',
+    message: 'View your usage stats',
+    autoDismiss: 0,
+    onDismiss: () => {
+      setTimeout(() => {
+        const isNaggingPath = getIsNaggingPath(history)
+        if (!isNaggingPath) return
+        emitNag(atmosphere, history)
+      }, NAG_EVERY)
+    },
+    action: {
+      label: 'View Usage',
+      callback: () => {
+        atmosphere.eventEmitter.emit('removeSnackbar', ({key}) => key === 'usage')
+        history.push(`/usage`)
+        SendClientSegmentEventMutation(atmosphere, 'Clicked usage snackbar CTA')
+      }
+    }
+  })
+  SendClientSegmentEventMutation(atmosphere, 'Sent usage snackbar')
+}
+
+const useUsageSnackNag = (insights: boolean) => {
+  const atmosphere = useAtmosphere()
+  const history = useHistory()
+  const isNaggingPath = getIsNaggingPath(history)
+  const isUserNaggable = isNaggingPath && insights
+  const {viewer} = useLazyLoadQuery<useUsageSnackNagQuery>(
+    graphql`
+      query useUsageSnackNagQuery($isUserNaggable: Boolean!) {
+        viewer {
+          domains @include(if: $isUserNaggable) {
```

Comment:
I know that the `domains` query is expensive, so rather than run this query for all users in the dashboard, I thought it'd be a good opportunity to use the `include` directive

Reply:
is it possible to avoid sending a request to the server if the user does not have the feature flag enabled?

Reply:
If the user does not have the `insights` feature flag enabled, `isUserNaggable` will be false, so the `domains` query will be ignored and won't be sent to the server. Are you thinking we should add the `@include` directive to `viewer` instead of `domains`?

As there are no other fields requested in `viewer`, if `domains` is ignored, there is no network request for `useUsageSnackNagQuery`:

![contains-query](https://user-images.githubusercontent.com/39854876/174578422-6ce75bb1-eb7d-4b1a-9663-dcd6432bc31d.png)

When `insights` is false:
![ignores-query](https://user-images.githubusercontent.com/39854876/174578436-33e1e325-311c-4017-aaf5-b76f70ef265a.png)

Reply:
WHOAAA. I had no idea that relay used the `@include` directive as a client-side filter! i thought it just naively sent it onto the server.
really cool find!